src/utils/messages.js

```diff
@@ -1,20 +1,28 @@
-const generateMessage = (username, text) => {
-    return {
-        username,
-        text,
-        createdAt: new Date().getTime(),
-    }
-}
+const User = require('../entities/user')
 
-const generateLocationMessage = (username, url) => {
-    return {
-        username,
-        url,
-        createdAt: new Date().getTime(),
-    }
-}
+const admin = new User({ username: 'admin' })
+
+const makeMessageGenerator =
+    (author, process = (p) => p, payloadKey = 'text') =>
+    (payload) => ({
+        username: author.username,
+        [payloadKey]: process(payload),
```

Comment:
Could you please clarify why is this object property is in square brackets?

Reply:
https://attacomsian.com/blog/javascript-computed-property-names